docs/pages/api/binaries/latest.ts

```diff
@@ -0,0 +1,73 @@
+import type { NextRequest } from "next/server";
+
+const REGISTRY = "https://registry.npmjs.org";
+const SUPPORTED_PACKAGES = new Set(["turbo"]);
+const TAG = "latest";
+
+const ERRORS = {
+  NOT_FOUND: {
+    status: 404,
+    error: "Not Found",
+  },
+  INTERNAL_SERVER_ERROR: {
+    status: 500,
+    error: "Internal Server Error",
+  },
+};
+
+export default async function handler(req: NextRequest) {
+  const { searchParams } = new URL(req.url);
+
+  // validate input if we have it
+  const requestedPackage = searchParams.get("package") || "turbo";
+  if (!SUPPORTED_PACKAGES.has(requestedPackage)) {
+    const { status, error } = ERRORS.INTERNAL_SERVER_ERROR;
+    return new Response(
+      JSON.stringify({
+        package: requestedPackage,
+        version: null,
+        error,
+      }),
+      {
+        status,
+      }
+    );
+  }
+
+  try {
+    const result = await fetch(`${REGISTRY}/${requestedPackage}/${TAG}`);
+    const json = await result.json();
+    return new Response(
+      JSON.stringify({
+        package: requestedPackage,
+        version: json.version,
+      }),
+      {
+        status: 200,
+        headers: {
+          "content-type": "application/json",
+          // cache for one day, and revalidate every hour
+          "cache-control":
+            "public, s-maxage=86400, stale-while-revalidate=3600",
+        },
+      }
+    );
+  } catch (e) {
+    console.error(e);
+    const { status, error } = ERRORS.INTERNAL_SERVER_ERROR;
+    return new Response(
+      JSON.stringify({
+        package: requestedPackage,
+        version: null,
+        error,
+      }),
+      {
+        status,
+      }
+    );
```

Comment:
Can we make this `function send500()` or something? We do it twice, one with a `console.error` one without. I kinda want it both places and that made me go "hmm, extract?"